ngi_reports/common/ign_sample_report.py

```diff
@@ -274,11 +274,17 @@ def parse_picard_metrics(self):
     def create_aggregate_statistics (self):
 
         def create_header (samples):
-            return samples.values()[0].keys()
+            # take the union of all keys for the header row
+            return sorted(
+                list(
+                    set(
+                        [key \
+                        for sample in samples.values() \
+                        for key in sample.keys()])))
```

Comment:
The construct is called list comprehension and I believe it can have an impact on performance, although you'd probably have to work with really large lists before any effect would be noticeable. I'm mostly thinking of it as personal preference.